iceoryx_examples/automotive_soa/include/owl/kom/event_subscriber_uds.inl

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_EXAMPLES_AUTOMOTIVE_SOA_EVENT_SUBSCRIBER_UDS_INL
+#define IOX_EXAMPLES_AUTOMOTIVE_SOA_EVENT_SUBSCRIBER_UDS_INL
+
+#include "owl/kom/event_subscriber_uds.hpp"
+
+namespace owl
+{
+namespace kom
+{
+template <typename T>
+inline EventSubscriberUds<T>::EventSubscriberUds(const core::String&,
+                                                 const core::String& instance,
+                                                 const core::String&) noexcept
+    : m_uds(std::move(iox::posix::UnixDomainSocket::create(instance, iox::posix::IpcChannelSide::SERVER)
+                          .or_else([](auto&) {
+                              std::cout << "Failed to create UNIX domain socket!" << std::endl;
+                              std::terminate();
+                          })
+                          .value()))
+{
+}
+
+template <typename T>
+inline void EventSubscriberUds<T>::Subscribe(std::size_t) noexcept
+{
+    // Subscribe not supported with UDS
+}
+
+template <typename T>
+void EventSubscriberUds<T>::Unsubscribe() noexcept
+{
+    // Unsubscribe not supported with UDS
+}
+
+template <typename T>
+template <typename Callable>
+core::Result<size_t> inline EventSubscriberUds<T>::GetNewSamples(Callable&& callable,
+                                                                 size_t maxNumberOfSamples) noexcept
+{
+    IOX_DISCARD_RESULT(maxNumberOfSamples);
+
+    core::Result<size_t> numberOfSamples{1};
+
+    auto samplePtr = std::make_unique<SampleType>();
+    SampleType& sample = *samplePtr;
+
+    std::string tempBuffer;
+
+    // Receive the first (up to) 4095 Bytes
+    m_uds.receive().and_then([&](auto& msg) { tempBuffer.append(msg); }).or_else([](auto&) {
+        std::cerr << "Error occurred while receiving UNIX domain socket!" << std::endl;
+    });
+
+    if (tempBuffer.size() == 0)
+    {
+        return 0;
+    }
+
+    // Deserialize the counter
+    sample.counter =
+        (static_cast<uint32_t>(tempBuffer[3]) & 0xFF) << 24 | (static_cast<uint32_t>(tempBuffer[2]) & 0xFF) << 16
```

Comment:
Undefined behavior, you never checked that `tempBuffer` has the correct size, it must at least have a size of 15 I think.